### ISSUE TITLE
%!v(PANIC=Error method: runtime error: slice bounds out of range [2:1])

### ISSUE THREAD

**Issue by kitagry** [goccy/go-yaml#64]
Above error occurs when I write code below.
I found this occurs at `printer.goL250`. But, I don't understand how to fix it correctly. Sorry.

https://github.com/goccy/go-yaml/blob/3032e43e20e487986fc151bf2f67f4dda4f5cd6f/printer/printer.go#L249-L251

```go
type C struct {
	B string `yaml:"b"`
}
                           
var v struct {
	A C `yaml:"complicated"`
}

const src = `---
complecated:
  b: string
`

err := yaml.NewDecoder(strings.NewReader(src), yaml.DisallowUnknownField()).Decode(&v)
fmt.Printf("%v\n", err)

// OUTPUT:
// [2:1] unknown field "complecated"
//        1 | ---
//     >  2 | complecated:
//           ^
//        3 |   b: string
```

**Comment by goccy** [comment]
@kitagry 

Thanks for the reporting !
I fixed with #67  and released as v1.1.7

**Comment by kitagry** [comment]
Thanks @goccy !!
